src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -40,6 +41,28 @@ public:                                                           \
 
 namespace winrt::Microsoft::Terminal::Control::implementation
 {
+    struct SelectionColor : SelectionColorT<SelectionColor>
+    {
+        SelectionColor() = default;
+        WINRT_PROPERTY(uint32_t, TextColor);
+
+    public:
+        til::color Color() const
+        {
+            if (_TextColor & 0xff000000)
+            {
+                // We indicate that this is an indexed color by setting alpha to 1:
+                return til::color(gsl::narrow_cast<uint8_t>(_TextColor), 0, 0, 1);
+            }

```

Comment:
A quick, non-scientific check (setting a breakpoint and looking at the stack of the first dozen or so callers), I found that true colors seem only to be finalized somewhere in the guts of some "RenderSettings" stuff. Even if ControlCore has access to the render settings, I think it would feel odd to be so eager about realizing the end value of an indexed color (and also I wonder if these SelectionColor objects could be used in multiple places, with different color tables; I just have no idea).

But I totally hear you; this is pretty ugly. I was able to smooth it out a tad, while still retaining it as an indexed color--I just needed to express this in terms of TextColor, which natively supports indexed colors. I'm not sure why I didn't before--maybe I was trying to avoid rocking the boat by including a new header here, or maybe I just didn't make it far enough down the thought-road.